### PR TITLE
make MsQuicStatusCodes platform specific

### LINKED ISSUE #53506: QUIC: update status codes for unix platforms
https://github.com/microsoft/msquic/pull/1645 changed error codes for platforms other than Windows. 
While the goal seems to be better alignment with posix e.g. all the codes map to errno values, the actual values differ (and had in the past) among platforms. 
For example the `EOVERFLOW` is 75 on Linux but 84 on macOS. 

We should probably ether re-factor current code to have real PAL files and include one per platform instead of 
```c#
internal static uint Pending => OperatingSystem.IsWindows() ? Windows.Pending : Posix.Pending;
```

or we should map msquic codes to portable .NET enum.

### PR DESCRIPTION
**Pull request by wfurt**
https://github.com/microsoft/msquic/pull/1645/ introduced new status codes on non-windows platforms and this is reaction to it. Our old split to windows/posix no does not work since the actual errno values differ among OSes. 
The assumption about FreeBSD is rather weak but we can fix it later since MsQuic does not wok on FreeBSD anyway. 

fixes #53506
cc: @ThadHouse @nibanks

### CODE REVIEW

File: src/libraries/System.Net.Quic/src/System.Net.Quic.csproj [hunk 1]
```diff
@@ -15,7 +15,16 @@
     <Compile Include="System\Net\Quic\Implementations\Mock\*.cs" />
     <Compile Include="System\Net\Quic\Implementations\MsQuic\*.cs" />
     <Compile Include="System\Net\Quic\Implementations\MsQuic\Internal\*.cs" />
-    <Compile Include="System\Net\Quic\Implementations\MsQuic\Interop\*.cs" />
+    <Compile Include="System\Net\Quic\Implementations\MsQuic\Interop\MsQuicAlpnHelper.cs" />
```
Comment (ManickaP) [inline src/libraries/System.Net.Quic/src/System.Net.Quic.csproj:18]: Is this change of csproj on purpose?

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/MsQuicStatusCodes.Windows.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        internal const uint ConnectionTimeout = 0x80410006;
+        internal const uint ConnectionIdle = 0x80410005;
+        internal const uint HostUnreachable = 0x800704D0;
+        internal const uint InternalError = 0x80410003;
+        internal const uint ConnectionRefused = 0x800704C9;
+        internal const uint ProtocolError = 0x80410004;
+        internal const uint VerNegError = 0x80410001;
+        internal const uint TlsError = 0x80072B18;
+        internal const uint UserCanceled = 0x80410002;
+        internal const uint AlpnNegotiationFailure = 0x80410007;
```
Comment (ManickaP) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/MsQuicStatusCodes.Windows.cs:29]: Missing `StreamLimit` on purpose?

File: src/libraries/System.Net.Quic/src/System.Net.Quic.csproj (hunk 1 above)
Comment (wfurt) [reply src/libraries/System.Net.Quic/src/System.Net.Quic.csproj:18]: yes. The wildcard would always include all flavors. With the change we include common files here explicitly and then one specific for each platform.

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/MsQuicStatusCodes.Windows.cs (hunk 2 above)
Comment (wfurt) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/MsQuicStatusCodes.Windows.cs:29]: no. This was copy of existing code. I did not look for completeness nor did I check/change  the hResult values.

Comment (wfurt) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/MsQuicStatusCodes.Windows.cs:29]: updated.

**Review by ManickaP — APPROVED** [review]
Thanks!

**Review by CarnaViire — APPROVED** [review]

**Review by nibanks — APPROVED** [review]

**Review by scalablecory — APPROVED** [review]
This has been bugging me for a while, thanks :)

**Pull request merged**